lib/shared/addon/utils/util.js

```diff
@@ -345,6 +345,26 @@ export function isBadTld(name) {
   }
 }
 
+export function formatEndpoint(endpoint) {
```

Comment:
Can you use the parser in lib/shared/addon/utils/parse-uri.js instead of adding another one?